sdk/digitaltwins/digital-twins-core/src/digitalTwinsClient.ts

```diff
@@ -176,26 +176,18 @@ export class DigitalTwinsClient {
    *
    * @param digitalTwinId The Id of the digital twin to create or update.
    * @param digitalTwinJson The application/json digital twin to create.
-   * @param enableUpdate If true then update of an existing digital twin is enabled.
-   * @param options The operation options
+   * @param options Extended operation options including
+   *  digitalTwinsAddOptions.ifNoneMatch: Only perform the operation if the entity does not already exist.
    * @returns The created application/json digital twin and the http response.
    */
   public upsertDigitalTwin(
     digitalTwinId: string,
     digitalTwinJson: string,
-    options: OperationOptions = {}
+    options: DigitalTwinsAddOptionalParams = {}
   ): Promise<DigitalTwinsAddResponse> {
-    const digitalTwinsAddOptionalParams: DigitalTwinsAddOptionalParams = options;
-    const span = this.createSpan(
-      "DigitalTwinsClient-upsertDigitalTwin",
-      digitalTwinsAddOptionalParams
-    );
+    const span = this.createSpan("DigitalTwinsClient-upsertDigitalTwin", options);
```

Comment:
I notice your `createSpan` is not correct, since it doesn't properly update the `SpanOptions` that are passed into the generated client. This causes the span graph to have two siblings: the core-http request span and this convenience span, instead of the http span being parented properly.

Also I don't think the diagnostics namespace should be "DigitalTwinsCore", but rather just "DigitalTwins" as that is the actual namespace.

Reply:
Fixed.

Reply:
Fixed